src/cli_plugin/list/list.js

```diff
@@ -1,13 +1,28 @@
-import { statSync, readdirSync } from 'fs';
+import { statSync, readdirSync, readFileSync, accessSync, R_OK } from 'fs';
 import { join } from 'path';
 
+function readable(path) {
```

Comment:
For 5.0+, plugins are required to have `package.json` (the installer will without that file).  You should be able to assume that they exist, and a thrown error would be appropriate if the file didn't exist or was not readable.